webapp/loggedin/surveyViews/nodeDefEdit/nodeDefEdit.js

```diff
@@ -23,60 +25,48 @@ import * as NodeDefEditState from './nodeDefEditState'
 import { putNodeDefProp } from './../../../survey/nodeDefs/actions'
 import { closeNodeDefEdit } from './actions'
 
-class NodeDefEdit extends React.Component {
-
-  constructor (props) {
-    super(props)
-
-    this.state = {
-      editingCategory: false,
-      editingTaxonomy: false
-    }
-  }
-
-  close () {
-    this.props.closeNodeDefEdit()
-  }
-
-  render () {
-    const {
-      nodeDef, nodeDefParent,
-      nodeDefKeyEditDisabled, nodeDefMultipleEditDisabled,
-      displayAsEnabled, displayInEnabled,
-      putNodeDefProp,
-      canUpdateCategory
-    } = this.props
-
-    const {
-      editingCategory,
-      editingTaxonomy,
-    } = this.state
-
-    return nodeDef
-      ? (
-        <div className="node-def-edit">
-          {
-            editingCategory ?
-              (
-                <CategoriesView
-                  canSelect={canUpdateCategory}
-                  onSelect={category => putNodeDefProp(nodeDef, NodeDef.propKeys.categoryUuid, Category.getUuid(category))}
-                  selectedItemUuid={NodeDef.getCategoryUuid(nodeDef)}
-                  onClose={() => this.setState({ editingCategory: false })}/>
-              )
-              : editingTaxonomy ?
-              (
+const NodeDefEdit = props => {
+  const {
+    nodeDef, nodeDefParent,
+    nodeDefKeyEditDisabled, nodeDefMultipleEditDisabled,
+    displayAsEnabled, displayInEnabled,
+    putNodeDefProp,
+    canUpdateCategory,
+    closeNodeDefEdit,
+  } = props
+
+  const { i18n } = useContext(AppContext)
+
+  const [ editingCategory, setEditingCategory ] = useState(false)
+  const [ editingTaxonomy, setEditingTaxonomy ] = useState(false)
+
+  const close = () => closeNodeDefEdit()
+
+  return nodeDef
+    ? (
+      <div className="node-def-edit">
+        {
+          editingCategory
+            ? (
+              <CategoriesView
+                canSelect={canUpdateCategory}
+                onSelect={category => putNodeDefProp(nodeDef, NodeDef.propKeys.categoryUuid, Category.getUuid(category))}
+                selectedItemUuid={NodeDef.getCategoryUuid(nodeDef)}
+                onClose={() => setEditingCategory(false)}/>
+            )
+            : editingTaxonomy
+              ? (
                 <TaxonomiesView
                   canSelect={true}
                   onSelect={taxonomy => putNodeDefProp(nodeDef, NodeDef.propKeys.taxonomyUuid, Taxonomy.getUuid(taxonomy))}
                   selectedItemUuid={NodeDef.getTaxonomyUuid(nodeDef)}
-                  onClose={() => this.setState({ editingTaxonomy: false })}/>
+                  onClose={() => setEditingTaxonomy(false)}/>
               )
               : (
                 <div className="node-def-edit__container">
                   <TabBar tabs={[
                     {
-                      label: 'Basic',
+                      label: i18n.t('designerView.nodeDefEdit.basic'),
```

Comment:
nodeDefEdit is an independent component. this should be i18n.t('nodeDefEdit.basic')

Reply:
Ok, moved it to root